### PR TITLE
Add particle pivot field to emitter

### LINKED ISSUE #6126: Possibility to set particle pivot point in emitter.
**Is your feature request related to a problem? Please describe (REQUIRED):**
It will be useful to have pivot point for sprite (particle) in emitter

**Describe the solution you'd like (REQUIRED):**
image
or vector3 option
here:
image
(for `Animation` sprite)

**Describe alternatives you've considered (REQUIRED):**
using of many sprite duplicates with a lot of alpha

**Additional context (OPTIONAL):**

https://user-images.githubusercontent.com/2209596/138465981-7e591954-1a31-4899-ac4c-ec3ab9f071d8.mp4

### PR DESCRIPTION
**Pull request by Jhonnyg**
This PR adds a "pivot" field to the particle emitter that can be used to offset where the particle should be evaluated from

Fixes #6126

### CODE REVIEW

**Review by vlaaad — DISMISSED** [review]
I only reviewed the editor code, LGTM!

**Review by AGulev — DISMISSED** [review]
LGTM

**Review by britzl — DISMISSED** [review]
Looks good!

File: engine/particle/proto/particle/particle_ddf.proto [hunk 1]
```diff
@@ -173,6 +173,7 @@ message Emitter
     optional bool                   stretch_with_velocity  = 22 [default = false];
 
     optional float                  start_offset           = 23 [default = 0.0];
+    required dmMath.Point3          pivot                  = 24;
```
Comment (JCash) [inline engine/particle/proto/particle/particle_ddf.proto:176]: This is a bug, since you then cannot open old particle fx files

**Pull request merged**